core-web/libs/sdk/analytics/src/lib/shared/analytics.utils.ts

```diff
@@ -0,0 +1,79 @@
+import { ANALYTICS_SOURCE_TYPE } from './analytics.constants';
+import { DotAnalyticsConfig, PageViewEvent } from './analytics.model';
+
+/**
+ * Retrieves analytics attributes from a given script element.
+ *
+ * @return {DotAnalyticsConfig | null} - The analytics attributes or null if there are no valid attributes present.
+ */
+export const getDataAnalyticsAttributes = (location: Location): DotAnalyticsConfig => {
+    const script = getAnalyticsScriptTag();
+
+    const attributes = {
+        server: script.getAttribute('data-analytics-server') || location.href,
+        debug: script.hasAttribute('data-analytics-debug'),
+        autoPageView: script.hasAttribute('data-analytics-auto-page-view'),
+        key: script.getAttribute('data-analytics-key') || ''
+    };
+
+    return attributes;
+};
+
+/**
+ * Retrieves the analytics script tag from the document.
+ *
+ * @returns {HTMLScriptElement} - The analytics script tag.
+ */
+export const getAnalyticsScriptTag = (): HTMLScriptElement => {
+    const scripts = document.querySelector('script[data-analytics-server]');
+
+    if (!scripts) {
+        throw new Error('Analytics script not found');
+    }
+
+    return scripts as HTMLScriptElement;
+};
+
+/**
+ * Creates the data for a page view event.
+ *
+ * @param {string} event_type - The type of event.
+ * @param {Location} location - The location object.
+ * @returns {PageViewEvent} - The data for the page view event.
+ */
+export function createAnalyticsPageViewData(event_type: string, location: Location): PageViewEvent {
+    const urlParams = new URLSearchParams(location.search);
+    const utmParams = {
+        source: urlParams.get('utm_source') || undefined,
+        medium: urlParams.get('utm_medium') || undefined,
+        campaign: urlParams.get('utm_campaign') || undefined,
+        id: urlParams.get('utm_id') || undefined
+    };
+
+    const vpWidth = window.innerWidth;
+    const vpHeight = window.innerHeight;
+
+    const userLanguage = navigator.language;
+    const docEncoding = document.characterSet;
+
+    return {
+        event_type,
+        utc_time: new Date().toISOString(),
+        local_tz_offset: new Date().getTimezoneOffset(),
+        referer: document.referrer,
+        url: location.href,
+        page_title: document.title,
+        doc_path: location.pathname,
+        doc_host: location.hostname,
+        doc_protocol: location.protocol,
+        doc_hash: location.hash,
+        doc_search: location.search,
+        screen_resolution: `${window.screen.width}x${window.screen.height}`,
+        vp_size: `${vpWidth}x${vpHeight}`,
+        user_agent: navigator.userAgent,
+        user_language: userLanguage,
+        doc_encoding: docEncoding,
+        utm: utmParams,
+        src: ANALYTICS_SOURCE_TYPE
```

Comment:
I wonder if we can maintain the camelCase to maintain code standards that are Developer Friendly and if possible hide logic that uses this casing from the developer

I'm assuming the clients will use this object to work with Analytics API

Reply:
I need this in this exact way. If I change any of this, the backend must change it and affect the Clickhouse schema too.

Reply:
I saw this [comment](https://github.com/dotCMS/core/pull/30758#discussion_r1857380914) and that answered my question, marking as resolved.